src/games/strategy/triplea/ui/menubar/ExportMenu.java

```diff
@@ -168,27 +173,48 @@ public String getDescription() {
       // A small warning so users will not over-write a file,
       if (f.exists()) {
         final int choice =
-            JOptionPane.showConfirmDialog(null, "A file by that name already exists. Do you wish to over write it?",
+            JOptionPane.showConfirmDialog(frame, "A file by that name already exists. Do you wish to over write it?",
                 "Over-write?", JOptionPane.YES_NO_OPTION, JOptionPane.WARNING_MESSAGE);
         if (choice != JOptionPane.OK_OPTION) {
           return;
         }
       }
+
       final File file = f;
-      final Runnable t = () -> {
-        if (saveScreenshot(node, file, frame, gameData)) {
-          JOptionPane.showMessageDialog(null, "Map Snapshot Saved", "Map Snapshot Saved",
-              JOptionPane.INFORMATION_MESSAGE);
+      final SwingWorker<?, ?> task = new SwingWorker<Void, Void>() {
+        @Override
+        protected Void doInBackground() throws Exception {
+          saveScreenshot(node, file, frame, gameData);
+          return null;
+        }
+
+        @Override
+        protected void done() {
+          try {
+            get();
+            SwingUtilities.invokeLater(() -> {
+              JOptionPane.showMessageDialog(frame, "Map Screenshot Saved", "Map Screenshot Saved",
+                  JOptionPane.INFORMATION_MESSAGE);
+            });
+          } catch (final ExecutionException e) {
+            SwingUtilities.invokeLater(() -> {
+              JOptionPane.showMessageDialog(frame, e.getCause().getMessage(), "Error Saving Screenshot",
+                  JOptionPane.OK_OPTION);
+            });
+          } catch (final InterruptedException e) {
+            Thread.currentThread().interrupt();
```

Comment:
I believe Goetz's gist is

- Allow `InterruptedException` to propagate up the call stack, if possible.
- If it's not possible to propagate the `InterruptedException`, re-interrupt the thread so the fact that the thread has been interrupted will not be lost.

I think these recommendations go back to how Goetz et al. wanted a canonical mechanism to cancel background tasks when they were designing `java.util.concurrent` many years ago.

Note that wrapping an `InterruptedException` in an unchecked exception would not work for code (e.g. `Executor` implementations) that might be acting on the thread interruption status, as the mere act of catching an `InterruptedException` (and not re-throwing it as an `InterruptedException`) clears the thread's interruption status (hence, why Goetz suggests re-interrupting the thread).

I agree with your idea for introducing a library function to encapsulate the preferred behavior.  When you've decided what that behavior should be, let me know, and I'll extract that as part of this PR, if you like.